soc/hps_soc.py

```diff
@@ -152,11 +151,14 @@ def setup_ram(self, size):
         self.bus.add_slave("sram_lram", self.lram.bus, region)
         self.bus.add_region("sram", region)
 
+    # define the "arena" region even if it's length-zero
```

Comment:
I guess we need this to avoid errors from the linker script referencing a non-existent region when the option is turned off, right?

Reply:
Exactly.   I couldn't another way to do it (without doing something more complicated such as running the linker script through a preprocessor).